mobile/lib/components/utility/image_with_overlay.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:radioaktywne/extensions/build_context.dart';
+
+/// Widget representing an image with overlaid text on the bottom of the image.
+class ImageWithOverlay extends StatelessWidget {
+  const ImageWithOverlay({
+    super.key,
+    this.child,
+    required this.thumbnailPath,
+    this.imageConstructor = Image.network,
+    this.titleOverlay,
+    required this.isLoading,
+    this.titleOverlayPadding = const EdgeInsets.all(4),
```

Comment:
just for clarity, the constructor parameters should be in order, e.g.: super first, then required, then with no value, and last with default value